exercises/practice/grep/test-grep.bats

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/env bats
+load bats-extra
+
+setup() {
+    # note that the redirection `<<-` (with the hyphen) allows
+    # leading _tabs_ at the start of lines to improve readability:
+    # don't change the tabs to spaces.
+
+    cat > iliad.txt <<-END_ILIAD
+	Achilles sing, O Goddess! Peleus' son;
+	His wrath pernicious, who ten thousand woes
+	Caused to Achaia's host, sent many a soul
+	Illustrious into Ades premature,
+	And Heroes gave (so stood the will of Jove)
+	To dogs and to all ravening fowls a prey,
+	When fierce dispute had separated once
+	The noble Chief Achilles from the son
+	Of Atreus, Agamemnon, King of men.
+	END_ILIAD
+
+    cat > midsummer-night.txt <<-END_MIDSUMMER
+	I do entreat your grace to pardon me.
+	I know not by what power I am made bold,
+	Nor how it may concern my modesty,
+	In such a presence here to plead my thoughts;
+	But I beseech your grace that I may know
+	The worst that may befall me in this case,
+	If I refuse to wed Demetrius.
+	END_MIDSUMMER
+
+    cat > paradise-lost.txt <<-END_PARADISE
+	Of Mans First Disobedience, and the Fruit
+	Of that Forbidden Tree, whose mortal tast
+	Brought Death into the World, and all our woe,
+	With loss of Eden, till one greater Man
+	Restore us, and regain the blissful Seat,
+	Sing Heav'nly Muse, that on the secret top
+	Of Oreb, or of Sinai, didst inspire
+	That Shepherd, who first taught the chosen Seed
+	END_PARADISE
+}
+
+teardown() {
+    rm -f iliad.txt midsummer-night.txt paradise-lost.txt
```

Comment:
Is the `-f` needed?

Reply:
It just prevents non-zero exit status in case one of the files is missing.